test/associative_test_data.hpp

```diff
@@ -100,10 +100,16 @@ void ptr_set_test()
     
     T* t = new T;
     c.insert( c.end(), t );    
-    c.insert( c.end(), std::auto_ptr<T>( new T ) );
     c.insert( new T ); 
+#ifndef BOOST_NO_AUTO_PTR        
+    c.insert( c.end(), std::auto_ptr<T>( new T ) );
     std::auto_ptr<T> ap( new T );
     c.insert( ap );
+#else
```

Comment:
Since we're providing the `unique_ptr` overloads when `BOOST_NO_CXX11_SMART_PTR` is not defined, we should probably test them under the same condition, instead of only when `BOOST_NO_AUTO_PTR` is set.

Reply:
Right, but that doesn't work out in all cases: in return values you only have either/or exclusively. Have a peek at line 139:
```
#ifndef BOOST_NO_AUTO_PTR
    std::auto_ptr<C> ap2        = c.release();
#else
    std::unique_ptr<C> up2      = c.release();
#endif
```
Sigh - this testing stuff will become tedious then...

Reply:
Yes, you're right. Perhaps just do it for the cases where it can be done easily, and leave the tedious ones as is?